hera_qm/xrfi.py

```diff
@@ -277,512 +648,203 @@ def xrfi_run(indata, args, history):
     observations. Each set of flagging will be stored, as well as compressed versions.
     """
     if indata is None:
-        if (args.model_file is None) and (args.calfits_file is None):
-            raise AssertionError('Must provide at least one of: filename, '
+        if (model_file is None) and (calfits_file is None):
+            raise AssertionError('Must provide at least one of: indata, '
                                  'model_file, or calfits_file.')
-        warnings.warn('indata is none, not flagging on any data visibilities.')
-    elif isinstance(indata, UVData):
+        warnings.warn('indata is None, not flagging on any data visibilities.')
+    elif issubclass(indata.__class__, UVData):
         uvd = indata
-        if len(args.filename) == 0:
+        if filename is None:
             raise AssertionError('Please provide a filename to go with UVData object. '
                                  'The filename is used in conjunction with "extension" '
                                  'to determine the output filename.')
         else:
-            if isinstance(args.filename, str):
-                filename = args.filename
-            else:
-                filename = args.filename[0]
+            if not isinstance(filename, str):
+                raise ValueError('filename must be string path to file.')
     else:
-        # make sure we were given files to process
-        if len(indata) == 0:
-            if (args.model_file is None) and (args.calfits_file is None):
-                raise AssertionError('Must provide at least one of: filename, '
-                                     'model_file, or calfits_file.')
-            indata = None
-            warnings.warn('indata is none, not flagging on any data visibilities.')
-        elif len(indata) > 1:
-            raise AssertionError('xrfi_run currently only takes a single data file.')
+        filename = indata
+        uvd = UVData()
+        if infile_format == 'miriad':
```

Comment:
UVData objects do provide a read() function which tries to discern the type of the input, though we may still want something like this when we get into partial I/O